code/Middleware/DebugBarMiddleware.php

```diff
@@ -94,6 +95,7 @@ protected function afterRequest(HTTPRequest $request, HTTPResponse $response)
             exit();
         }
 
+        Deprecation::outputNotices();
```

Comment:
Normally deprecation notices are output in a shutdown function, which runs well after this middleware has output the debugbar.
This line ensures the deprecation notices which have been noted up to this point are logged out and collected by the `MonologCollector`.